tsdb/compact.go

```diff
@@ -541,6 +545,130 @@ func (c *LeveledCompactor) compact(dest string, dirs []string, open []*Block, sh
 	return nil, errs.Err()
 }
 
+// CompactOOOWithSplitting splits the input OOO Head into shardCount number of output blocks
+// per possible block range, and returns slice of block IDs. In result[i][j],
+// 'j' corresponds to the shard index while 'i' corresponds to a single time range of blocks.
```

Comment:
```suggestion
// 'i' corresponds to a single time range of blocks while 'j' corresponds to the shard index.
```

I'd expect to read the index keys in order

Reply:
Thanks, will update this